lute/backup/service.py

```diff
@@ -20,6 +22,54 @@ class BackupException(Exception):
     """
 
 
+class DatabaseBackupFile:
+    """
+    A representation of a lute backup file to hold metadata attributes.
+    """
+
+    def __init__(self, filepath: os.PathLike):
+        if not os.path.exists(filepath):
+            raise BackupException(f"No backup file at {filepath}.")
+
+        name = os.path.basename(filepath)
+        if not re.match(r"(manual_)?lute_backup_", name):
+            raise BackupException(f"Not a valid lute database backup at {filepath}.")
+
+        self.filepath = filepath
+        self.name = name
+        self.is_manual = self.name.startswith("manual_")
+
+    def __lt__(self, other):
+        return self.last_modified < other.last_modified
+
+    @property
+    def last_modified(self) -> datetime:
```

Comment:
Question for you: I haven't used return types for methods like you have here (`-> datetime`), but I like the concept.  In the past I was running into annoying circular import problems when I tried using these for custom types.  Any recommendation?  Is this information used only for type hints, or is it also picked up by pylint and other tools?

Reply:
Yes, the typing system really helps to jump to definition and see references. When you run into cyclical imports, the typing allows you to specify it as a string.

eg. `-> datetime.datetime` is equivalent to `-> "datetime.datetime"`

The first would require the import but the second has it differed.

Reply:
It is also picked up by the linters and in VSCode it will underline blue when you write code that doesnt respect the types.

Reply:
Thanks for the info.  I haven't really set up a good editing environment yet, just emacs and treemacs, but it's good enough for me.  But the type hinting is good.